### PR TITLE
OP-1033 Make PORTABLE mode to start BOTH gui and api server

### PR DESCRIPTION
**Pull request by mizzioisf**
See OP-1033 - Ready for 1.12.1

### CODE REVIEW

File: oh-extra/oh-api.ps1
Comment (mwithi) [inline oh-extra/oh-api.ps1:1]: should it be `$script:OH_DIR="."` ?

Comment (mizzioisf) [reply oh-extra/oh-api.ps1:1]: No, it is bundled from the oh-extra dir, it works this way

**Comment by mwithi** [comment]
When it starts the first time it creates a wrong JWT token

`org.springframework.web.util.NestedServletException: Request processing failed; nested exception is io.jsonwebtoken.security.WeakKeyException: The signing key's size is 416 bits which is not secure enough for the HS512 algorithm.  The JWT JWA Specification (RFC 7518, Section 3.2) states that keys used with HS512 MUST have a size >= 512 bits (the key size must be greater than or equal to the hash output size).  Consider using the io.jsonwebtoken.security.Keys class's 'secretKeyFor(SignatureAlgorithm.HS512)' method to create a key guaranteed to be secure enough for HS512.  See https://tools.ietf.org/html/rfc7518#section-3.2 for more information.`

**Comment by mizzioisf** [comment]
> When it starts the first time it creates a wrong JWT token
> 
> `org.springframework.web.util.NestedServletException: Request processing failed; nested exception is io.jsonwebtoken.security.WeakKeyException: The signing key's size is 416 bits which is not secure enough for the HS512 algorithm. The JWT JWA Specification (RFC 7518, Section 3.2) states that keys used with HS512 MUST have a size >= 512 bits (the key size must be greater than or equal to the hash output size). Consider using the io.jsonwebtoken.security.Keys class's 'secretKeyFor(SignatureAlgorithm.HS512)' method to create a key guaranteed to be secure enough for HS512. See https://tools.ietf.org/html/rfc7518#section-3.2 for more information.`

fixed

File: oh-extra/oh-api.ps1 [hunk 1]
```diff
@@ -917,9 +904,9 @@ function start_api {
 
         cd "$OH_PATH/$OH_DIR" # workaround for hard coded paths
 
-	$JAVA_ARGS="-client -Xms64m -Xmx1024m -cp ./bin/openhospital-api-0.0.2.jar;./rsc;./static org.springframework.boot.loader.JarLauncher"
+	$JAVA_API_ARGS="-server -Xms64m -Xmx1024m -cp ./bin/openhospital-api-0.0.2.jar;./rsc;./static org.springframework.boot.loader.JarLauncher"
 
-	Start-Process -FilePath "$JAVA_BIN" -ArgumentList $JAVA_ARGS -Wait -NoNewWindow -RedirectStandardOutput "$OH_PATH/$LOG_DIR/$API_LOG_FILE" -RedirectStandardError "$OH_PATH/$LOG_DIR/$API_ERR_LOG_FILE"
+	Start-Process -FilePath "$JAVA_BIN" -ArgumentList $JAVA_API_ARGS -Wait -WindowStyle Hidden -RedirectStandardOutput "$OH_PATH/$LOG_DIR/$API_LOG_FILE" -RedirectStandardError "$OH_PATH/$LOG_DIR/$API_ERR_LOG_FILE"
```
Comment (mwithi) [inline oh-extra/oh-api.ps1]: Apparently this solve the problem of launching API+GUI

```suggestion
	Start-Process -FilePath "$JAVA_BIN" -ArgumentList $JAVA_API_ARGS -WindowStyle Hidden -RedirectStandardOutput "$OH_PATH/$LOG_DIR/$API_LOG_FILE" -RedirectStandardError "$OH_PATH/$LOG_DIR/$API_ERR_LOG_FILE"
```

Comment (mizzioisf) [reply oh-extra/oh-api.ps1]: Yes, it should work.

**Pull request merged**